cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -402,6 +402,54 @@ Status GroupToStruct(const GroupNode& node, int16_t max_def_level, int16_t max_r
   return Status::OK();
 }
 
+Status MapToSchemaField(const GroupNode& group, int16_t max_def_level,
+                        int16_t max_rep_level, SchemaTreeContext* ctx,
+                        const SchemaField* parent, SchemaField* out) {
+  if (group.field_count() != 1) {
+    return Status::NotImplemented(
+        "Only MAP-annotated groups with a single child can handled");
+  }
+  out->children.resize(1);
+  SchemaField* child_field = &out->children[0];
+  ctx->LinkParent(out, parent);
+  ctx->LinkParent(child_field, out);
+
+  const Node& key_val_node = *group.field(0);
+
+  if (!key_val_node.is_repeated()) {
+    return Status::NotImplemented(
```

Comment:
I don't think NotImplemented here is correct, this should be considered invalid input.  Consider providing the offending field name (and additional metadata if possible) in all these error messages, it will make determining errors easier.